### ISSUE TITLE
Effect created when sending trust to self

### ISSUE THREAD

**Issue by bartekn** [stellar/go#2121]
### What version are you using?

Branch: `release-horizon-v0.25.0`

### What did you do?

Running `horizon expingest verify-range` on pubnet history.

### What did you expect to see?

It seems that the legacy ingestion did not create effects for this operation: https://horizon.stellar.org/operations/3529780217516033

### What did you see instead?

The new system creates an effect (fields used in DB dump: `history_effects.history_operation_id, history_effects.order, type, details, history_accounts.address`):

```
3529780217516033,1,0,
{
    "limit": "922337203685.4775807",
    "asset_code": "JPY",
    "asset_type": "credit_alphanum4",
    "asset_issuer": "GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM"
}
,GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM
```
Please note that effect `type` is equal `0`.

**Comment by bartekn** [comment]
Looks like the old system has a special `if` for such situation:
```go
		// NOTE:  when an account trusts itself, the transaction is successful but
		// no ledger entries are actually modified, leading to an "empty meta"
		// situation.  We simply continue on to the next operation in that scenario.
		if err == meta.ErrMetaNotFound {
			return
		}
```

**Comment by bartekn** [comment]
Two more instances of this issue found in ops: 16600645599498241, 16611254168719361.

**Comment by abuiles** [comment]
Looking into this.

**Comment by abuiles** [comment]
Fixed in #2124
